### DISCUSSION TITLE
How to turn off GoPro between time lapse photos?

### DISCUSSION THREAD

**Discussion by CHEETHAM1** [gopro/labs#916] [Q&A]
Hi all, very new to this LABS thing. I'm trying to capture a long term time lapse using a GoPro 11 and 12. I'd like to have a 24/7 night lapse running at 5 min intervals (possibly shorter or longer pending testing). The most critical thing I need at this stage is the ability to turn the cam off, then wake before next photo. Is this a custom code? I cannot find the option in the QR Control app or code creator page. Thanks

**Comment by dnewman-gpsw** [comment]
The basics are you have to put the camera asleep between actions.  Example:  !60N sleeps the camera for 60 seconds. mP!S this is mode Photo then shutter press.  !1R is repeat in 1 second.  Now you should see how !60NmP!S!1R works

**Reply by CHEETHAM1** [reply]
Thank you. Is there a tutorial on how to generate and apply these codes?
I can see there are tick box web page and app methods to generate codes, then there are codes produced manually by those with more experience.
Do you apply multiple codes to achieve what you want, or one long code?
Thanks

**Reply by dnewman-gpsw** [reply]
Learn the command language and use any QR code generator that has a text input.  Docs: https://gopro.github.io/labs/control/tech/

**Reply by CHEETHAM1** [reply]
![IMG_9467](https://github.com/gopro/labs/assets/170095402/4494ba26-731b-4439-9d89-1c27613567f8)
When I originally began playing with the code above there was a 10 second countdown before the photo, now it’s changed to 20/21seconds. Is there anyway this time can be reduced for better battery life.

**Reply by dnewman-gpsw** [reply]
What command did this? Sleeps over 9 seconds, will typically shutdown the camera.  Share the QR Code or the command string (preferred),

**Reply by CHEETHAM1** [reply]
This is the command you suggested above: !60NmP!S!1R

**Reply by CHEETHAM1** [reply]
I also found the 9-10 second count down is when on gopro battery and when charging off USB cable it changes to 20/21 second countdown.

**Reply by dnewman-gpsw** [reply]
Okay try:  !60NQmP!S!1R   That will have less countdown.

**Reply by CHEETHAM1** [reply]
That command is working great thank you David. 
You seem to be very knowledgeable in this field, are you aware of a GoPro reliably running scripts like this for a long period (externally powered)? I find my 12 more consistent than my 11 which I’ve been testing with.

**Reply by dnewman-gpsw** [reply]
I agree HERO12 is the most reliable, but closely followed by H11.  I've run scripts like this for months with H12.  I travelling for 3 weeks, I left two H12 running, but I expect will be running in my return.  I've had one H12 running since the beginning of the year.  All with an internal battery, as the can improve the reliability for anything using the clock.

**Comment by CHEETHAM1** [comment]
Do you trial running the H12 with an external power supply? When I try without a battery in (previously worked on other models) it comes up with an error saying external power supply needs to be 5V/2A regardless of what power supply. I've tried 5V/3A and even increasing the voltage but the error either remains, or the GoPro stops working. The only way I can get it to run a schedule is with the battery in, but charging from the external supply.

**Comment by CHEETHAM1** [comment]
Also I assume from your long term trials that you are taking images (or other commands) very infrequently - Hourly, daily etc. Apart from time, what is the most "actions" you've successfully carried out. My intentions are to use these to capture a medium term time lapse which would require 2-10min intervals and run for several days. This would equal thousands of actions/photos.

**Comment by dnewman-gpsw** [comment]
The power error is your source or the cable it not reporting the current capacity of the power source.  You can fix this with *TUSB=1.

For intervals I've tested shorter more, as testing is faster.  I've tested H12 to about 10,000 starts.  It has not ever stopped.

**Reply by CHEETHAM1** [reply]
Thanks David, does that code go at the end of the current script?

**Reply by dnewman-gpsw** [reply]
No, it is a one time setting (extension).

**Comment by CHEETHAM1** [comment]
Hi again David, I’m still having issues getting these G12’s to run consistently. The longest they run for without randomly stopping is around 12hrs but cycle count can be 200-450 photos (2min. Intervals) This is powered by a battery pack, not internal battery. 
All I’m running after a RESET, is *TUSB=1 and !120NQmPN!S!1R
Any ideas?

**Reply by dnewman-gpsw** [reply]
The log file will help from MISC/qrlog.txt

Also having the interval battery installed will help, in addition to the external power.

**Reply by CHEETHAM1** [reply]
I will download the log files. 
I was running one of the G12’s with the internal battery in as well as external power and found it was getting hot. The battery itself had slightly swollen but still went in and out ok. This is with a voltage regulator outputting 5v / 2A.

**Reply by dnewman-gpsw** [reply]
There is no reason for the camera to get hot if it is only on for 10 seconds every two minutes.  It is good to start with a charged battery using the external battery charger.

**Reply by CHEETHAM1** [reply]
I agree. I don’t know why it was hot. Had two G12’s running parallel with same script. The internal battery one was warm and the other was fine. I removed the battery and it cooled down and kept running.

**Reply by CHEETHAM1** [reply]
Hi David, I've placed the logs in this folder. Both stopped again this morning while I was at work. G12#1 stopped at 0300am and G12#2 stopped at 0816am. 
https://drive.google.com/drive/folders/1giB-bX4YgpxOAQigti6mliXMCfHN4UTh?usp=sharing
Thanks for your help.
Perhaps it Night Photo related? I might go back to normal photo for testing as the scene I will be capturing should be well lit at night.

**Reply by dnewman-gpsw** [reply]
Night Photo can be issue if the exposures are very long, but that shouldn't be the case at these times.  Repeat with the battery inserted. These look typical to setting alarm events with no battery.

**Reply by CHEETHAM1** [reply]
I changed to day photos and restarted both this morning, one with an internal battery G12#2, one without G12#1, both power via USB C.
The one without a battery lasted half and hour and the one with lasted about 5hrs. My trials are getting worse?
Logs below.
https://drive.google.com/drive/folders/1giB-bX4YgpxOAQigti6mliXMCfHN4UTh?usp=drive_link

**Reply by dnewman-gpsw** [reply]
Just statistics, not getting worse.  It's puzzling why this isn't working for you.   Things to try.  *DBGL=1 for more debugging.  Factory reset.  Set Photo as the default mode.

**Reply by pete1414** [reply]
Does anyone have any updates on this?  My Gopro 10 experiences the exact same issue..

**Reply by dnewman-gpsw** [reply]
What issue?  That thread is clear as mud. Labs turning off between images is a core function and works well.  You likely have a different issue, so ask in your own thread.

**Reply by pete1414** [reply]
In my case, it does not work well, same as the OP.  It randomly stops.  Sometimes it works till the battery is dead, more often than not it stops with no cause.    
![image](https://github.com/user-attachments/assets/c63cf394-6bf8-4d1b-8f43-caf6ce4e691f)

**Reply by CHEETHAM1** [reply]
Yes same experience here. Random stoppages. I gave up after purchasing two new 12’s for this task.

**Reply by Espusen** [reply]
What Hero model are you using?

**Reply by Espusen** [reply]
I have been using Hero12s for TimeLapse for some time and they are working great. 
Hero10 and Hero9 Not so sadly.

**Reply by dnewman-gpsw** [reply]
@pete1414 I've run hero 12 for 7 months without an issue, so if you're not sharing commands or your logs, I cannot help you.

**Reply by pete1414** [reply]
> @pete1414 I've run hero 12 for 7 months without an issue, so if you're not sharing commands or your logs, I cannot help you.

I was just hoping the OP had found a solution,  but I am more than willing/happy to share anything with you,  if you are offering to help!    This is the script I have been testing with for the past 2 weeks, every morning  mPoD0oA0oV0oW0g0e0v0r4fwr0!S!120N!R     but the last 2 attempts, I used the script you shared above.    !60NQmP!S!1R (only thing I change is 60 to 120 to get a photo every 2 minutes.         I started it yesterday at 4:18pm, I checked on it this morning at 3:35am and there is only 47 photos.  When I turn on the gopro there is 85% battery and a message says 'START TIME 18:01:00 (14.6 hours) button to cancel'   so it seemed to operate fine for about 2 hours then some kind of time issue occurred causing it to think the next photo is due in in 14.6 hours instead of 2 minutes?   I am attaching my QR log, a photo of the screen when I turn on the gopro..     Photo mode is the default mode, wifi, gps, voice, raw photos are all disabled, auto power off is disabled, battery is a brand new Enduro battery, SD card is sandisk extreme, environment is indoors, just placed on a window while I am testing.   Issue is random, sometimes it'll run till battery dies, but more often than not, when I turn the camera on I get that message  START TIME .... some amount of hours.   The QR log last line is never the same..    I've tried formatting the card in the camera before several attempts that have also failed... I've tried formatting the card on my PC and have failures with both.  I've tried downloading a new copy of the gopro labs firmware and reflashed it.    I have contacted GoPro and the next step they suggested I install the official firmware, then install the labs firmware again.   Then they sent me over here.   I've only recently learned about  *DBGL=1 (in this thread)  but I did scan it before running the script so the qr log should be full.   It definitely has more information than I've ever seen before.   My next test, I'll do a factory reset, then scan the debug code, then the script.  
[qrlog.txt](https://github.com/user-attachments/files/18764824/qrlog.txt)

![20250212_032611](https://github.com/user-attachments/assets/dba8bd2f-fecb-4c63-8bef-551e9bbb8de5)

**Reply by Espusen** [reply]
I think David will shed some light on this. An error is showing on the log but I cannot make out what that is. 
min wondering if this is because it misses the time, then it will wait until it comes around again? Can you insert an “approximately X minutes” here David? 

I have an issue with mine often that this will happen after a program start, then it picks it up again the same time 24 hours later and runs for weeks again.

**Reply by dnewman-gpsw** [reply]
Thanks for this extra detail. This way it needed to be a new thread, the original post was talking about a different camera model, and got confused I had missed you had a HERO10. The log of course confirmed this. The camera before the HERO11 are unreliable at wake events. HERO12 and 13 are the best at it.  

This issue happen after this step 
[2025-02-11 18:01:54-delayed ] wake reason:RTC_ALARM 
[2025-02-11 18:01:55 100%  ] Woke with Alarm:[18:01m01]!NQmP!S!1R 
[2025-02-11 18:01:55 --%   ] Labs:Display Messages
then nothing

Should have looked like:
[2025-02-11 17:59:42-delayed ] wake reason:RTC_ALARM 
[2025-02-11 17:59:43 100%  ] Woke with Alarm:[17:59m01]!NQmP!S!1R 
[2025-02-11 17:59:43 87%   ] Labs:Display Messages 
[2025-02-11 17:59:44 87%   ] display 2000ms:Features activated: 
TUSB 
[2025-02-11 17:59:47 87%   ] Features activated: 
TUSB: 
[2025-02-11 17:59:47 87%   ] Labs:Main Start 
[2025-02-11 17:59:47 87%   ] Labs:H21.01.01.62.70 
[2025-02-11 17:59:47 87%   ] Labs LCD ID:195 
[2025-02-11 17:59:47 87%   ] Naked when 0:1 
[2025-02-11 17:59:47 87%   ] Labs:Initialize 
...

Have you ever found you have to do a random battery pull on HERO10? Camera won't turn on of off randomly? This same randomness is effecting the script from continuing. The need for battery pulls was eliminated with HERO12.

**Reply by pete1414** [reply]
I have not had to do a random battery pull, the only time it wouldn't power on for me is when the script was completely successful for 12+ hours and the battery was depleted.  but I really haven't had a chance to 'use' it yet, just this testing.    Sorry for the confusion, I don't know anything about newer models, and saw this thread was similar/identical to my issue and was hoping the OP had a solution.    Guess this is not something that will work for me, being an older model doesn't seem like the issue of not being reliable at waking will be addressed in labs or official firmware.   I do appreciate your time!

**Reply by dnewman-gpsw** [reply]
On older models just use the stock timelapse feature, with external power. And you want need Labs for that.
